calico/felix/devices.py

```diff
@@ -372,21 +372,31 @@ def remove_conntrack_flows(ip_addresses, ip_version):
         _log.debug("Removing conntrack rules for %s", ip)
         for direction in ["--orig-src", "--orig-dst",
                           "--reply-src", "--reply-dst"]:
-            try:
-                futils.check_call(["conntrack", "--family",
-                                   "ipv%s" % ip_version, "--delete",
-                                   direction, ip])
-            except FailedSystemCall as e:
-                if e.retcode == 1 and "0 flow entries" in e.stderr:
-                    # Expected if there are no flows.
-                    _log.debug("No conntrack entries found for %s/%s.",
-                               ip, direction)
+            remaining_attempts = 3
+            while remaining_attempts > 0:
+                remaining_attempts -= 1
+                try:
+                    futils.check_call(["conntrack", "--family",
+                                       "ipv%s" % ip_version, "--delete",
+                                       direction, ip])
+                except FailedSystemCall as e:
+                    if e.retcode == 1 and "0 flow entries" in e.stderr:
+                        # Success: there are no flows.
+                        _log.debug("No conntrack entries found for %s/%s.",
+                                   ip, direction)
+                        break
+                    if remaining_attempts == 0:
```

Comment:
Are you sure that it won't be useful to see all the exceptions - rather than just the last one?  Seeing all of them might eventually give us understanding about why conntrack calls are failing.